File: JK/80. Remove Duplicates from Sorted Array II.py
def removeDuplicates(nums):
    index = 1  
    count = 1  

    for i in range(1, len(nums)):
        if nums[i] == nums[i - 1]:
            count += 1
        else:
            count = 1
        
        if count <= 2:
            nums[index] = nums[i]
            index += 1               

    return index  


# nums = [1,1,1,2,2,3]
# removeDuplicates(nums)

File: JK/80. Remove Duplicates from Sorted Array II.c

#include <stdio.h>

int removeDuplicates(int* nums, int numsSize) {
    if (numsSize <= 2) {
        return numsSize;
    }
    
    int index = 1;  
    int count = 1;  

    for (int i = 1; i < numsSize; i++) {
        if (nums[i] == nums[i - 1]) {
            count++;
        } else {
            count = 1;
        }

        if (count <= 2) {
            nums[index] = nums[i];
            index++;
        }
    }

    return index;
}



int main() {
    int nums[] = {0,0,1,1,1,1,2,3,3};
    int numsSize = sizeof(nums) / sizeof(nums[0]);
    
    int newSize = removeDuplicates(nums, numsSize);
    
    printf("Array after removing duplicates: \n");
    for (int i = 0; i < newSize; i++) {
        printf("%d ", nums[i]);
    }
    printf("\n");
    
    return 0;
}